### PR TITLE
Feat/discover page

### PR DESCRIPTION
**Pull request by khubak**
TODO: 

* fix grid misusage
* utilize usememo hook

### CODE REVIEW

File: src/models/genre-new/index.ts
Comment (d-reader-josip) [inline src/models/genre-new/index.ts:1]: remove `genre-new`

**Review by d-reader-josip — CHANGES_REQUESTED** [review]

File: src/app/(authenticated)/discover/comics/page.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { DiscoverWrapper } from '@/components/discover/DiscoverWrapper'
+import { ComicsContent } from '@/components/discover/ComicsContent'
+
+export default function DiscoverComicsPage() {
+  return (
+    <DiscoverWrapper>
+      <ComicsContent />
```
[the remark this answers is not in the document]
Comment (khubak) [reply src/app/(authenticated)/discover/comics/page.tsx:8]: thoughts on custom ComicsContent component with slider/grid/row types?

File: src/components/discover/ComicIssuesContent.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                <div className='flex w-full justify-between'>
+                  <Text as='p' styleVariant='body-xsmall' className='text-gray-100'>
+                    EP {comicIssue.number}/{comicIssue.stats.totalIssuesCount}
+                  </Text>
+                  <PriceTag styleVariant='body-xsmall' colorfulIcon size={11} price={comicIssue.stats.price} />
+                </div>
+              )}
+            </div>
+          </div>
+        </Link>
```
[the remark this answers is not in the document]
Comment (khubak) [reply src/components/discover/ComicIssuesContent.tsx]: will do once figma for all the cards is ready

File: src/api/genre/queries/useFetchGenres.ts
Comment (d-reader-josip) [inline src/api/genre/queries/useFetchGenres.ts:1]: delete this

File: src/app/(authenticated)/discover/creators/page.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { DiscoverWrapper } from '@/components/discover/DiscoverPageWrapper'
+import { CreatorGrid } from '@/components/discover/CreatorGrid'
+
+export default function DiscoverComicsPage() {
```
Comment (d-reader-josip) [inline src/app/(authenticated)/discover/creators/page.tsx:5]: DiscoverCreatorsPage

File: src/components/discover/ComicGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { useDiscoverFilterStore } from '@/providers/DiscoverFilterStoreProvider'
+import { Comic } from '@/models/comic'
+import { RoutePath } from '@/enums/routePath'
+import { useFetchComics } from '@/api/comic/queries'
+import { useEffect, useMemo } from 'react'
+import { ShowMoreButton } from './ShowMoreButton'
+
+export const ComicGrid: React.FC = () => {
+  const storeComicParams = useDiscoverFilterStore((state) => state.comicParams)
+  const comicParams = useMemo(() => storeComicParams, [storeComicParams])
```
Comment (d-reader-josip) [inline src/components/discover/ComicGrid.tsx:15]: This is unnecessary

File: src/components/discover/ComicGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            </Text>
+            <div className='flex flex-col pl-2 mt-2 justify-center'>
+              <Text as='p' styleVariant='body-large' fontWeight='bold'>
+                {comic.title}
+              </Text>
+              <Text as='p' styleVariant='body-normal'>
+                {comic.creator ? 'by ' + comic.creator.name : ''}
+              </Text>
+            </div>
+          </Link>
```
Comment (d-reader-josip) [inline src/components/discover/ComicGrid.tsx:57]: This should be a Comic component

File: src/components/discover/ComicGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    <>
+      <div className='grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 1160:grid-cols-4 lg:grid-cols-5 xl:grid-cols-6 gap-6 pt-2'>
+        {comics.map((comic: Comic) => (
+          <Link
+            href={RoutePath.Comic(comic.slug)}
+            className='flex flex-col relative w-full h-[307px] rounded-xl hover:brightness-125 border border-grey-300 p-2'
+            key={comic.title}
+          >
+            <Image
+              alt={comic.title + ' Cover'}
```
Comment (d-reader-josip) [inline src/components/discover/ComicGrid.tsx:35]: put alt as empty string

File: src/components/discover/ComicIssueGrid.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+import { useEffect, useMemo } from 'react'
+import { ComicIssue } from '@/models/comicIssue'
+import { RoutePath } from '@/enums/routePath'
+import { PriceTag } from '../shared/tags/PriceTag'
+import { useFetchComicIssues } from '@/api/comicIssue/queries'
+import { ShowMoreButton } from './ShowMoreButton'
+
+export const ComicIssueGrid: React.FC = () => {
+  const storeComicIssueParams = useDiscoverFilterStore((state) => state.comicIssueParams)
+  const comicIssueParams = useMemo(() => storeComicIssueParams, [storeComicIssueParams])
```
Comment (d-reader-josip) [inline src/components/discover/ComicIssueGrid.tsx:16]: remove this useMemo

File: src/components/discover/DiscoverPageWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import { DiscoverQueryBar } from './DiscoverQueryBar'
+import { cn } from '@/lib/utils'
+import { DISCOVER_PAGE_TABS } from '@/constants/tabs'
+import { DiscoverFilterStoreProvider } from '@/providers/DiscoverFilterStoreProvider'
+import { fetchMe } from '@/app/lib/api/user/queries'
+import { Navigation } from '../layout/Navigation'
+
+type Props = React.PropsWithChildren & { mainClassName?: string }
+
+export const DiscoverWrapper: React.FC<Props> = async ({ children, mainClassName }) => {
```
Comment (d-reader-josip) [inline src/components/discover/DiscoverPageWrapper.tsx:12]: Rename to DiscoverPageWrapper

File: src/components/discover/DiscoverPageWrapper.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      <Navigation me={me} hideSearch />
+      <main
+        className={cn(
+          'flex flex-col w-full h-full items-center mt-20 md:mt-16 p-4 md:p-6 lg:p-8 xs:pb-24 sm:pb-24 md:pb-24 lg:pb-24 flex-1',
+          mainClassName
+        )}
+      >
+        <div className={cn('flex flex-col max-w-screen-xl w-full gap-3')}>
+          <DiscoverFilterStoreProvider>
+            <Tabs label={'Discover'} tabs={DISCOVER_PAGE_TABS} />
```
Comment (d-reader-josip) [inline src/components/discover/DiscoverPageWrapper.tsx:26]: remove "{" and "}" brackets

File: src/components/discover/ShowMoreButton.tsx
Comment (d-reader-josip) [inline src/components/discover/ShowMoreButton.tsx:1]: Ask Matan to  add a "show more" button and explain the infinite scroll behavior

File: src/components/shared/Tabs.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+export const Tabs: React.FC<Props> = ({ label, tabs }) => {
+  const selectedTab = usePathname()
+
+  return (
+    <div className='flex flex-col w-full relative '>
+      <Text
+        as='h1'
+        styleVariant='primary-heading'
+        className='absolute -top-6 left-2 text-grey-100 text-transparent uppercase bg-gradient-to-b from-grey-400 via-grey-600 via-60% to-transparent bg-clip-text'
```
Comment (d-reader-josip) [inline src/components/shared/Tabs.tsx:23]: Switch from absolute to flex (with negative margin-top on the component below)

File: src/constants/tabs.ts [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  slug: string
+  name: string
+  url: string
+}
+
+export const DISCOVER_PAGE_TABS: Tab[] = [
+  {
+    slug: 'comics',
+    name: 'Comics',
+    url: '/discover/comics',
```
Comment (d-reader-josip) [inline src/constants/tabs.ts:11]: use `RoutePath.DiscoverComics`

File: src/providers/DiscoverFilterStoreProvider.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+export type DiscoverFilterStoreApi = ReturnType<typeof createDiscoverQueryStore>
+export const DiscoverFilterStoreContext = createContext<DiscoverFilterStoreApi | undefined>(undefined)
+
+const fetchData = async (updateCompleteGenresList: (genres: Genre[]) => void) => {
+  try {
+    const completeGenresList = await fetchGenres({})
+    updateCompleteGenresList(completeGenresList)
+  } catch (error) {
+    console.error('Error fetching genres: ', error)
+  }
```
Comment (d-reader-josip) [inline src/providers/DiscoverFilterStoreProvider.tsx:19]: I don't think this catch is necessary tbh?

File: src/providers/DiscoverFilterStoreProvider.tsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+import { fetchGenres } from '@/app/lib/api/genre/queries'
+import { Genre } from '@/models/genre'
+
+export const store = createDiscoverQueryStore(defaultInitState)
+export type DiscoverFilterStoreApi = ReturnType<typeof createDiscoverQueryStore>
+export const DiscoverFilterStoreContext = createContext<DiscoverFilterStoreApi | undefined>(undefined)
+
+const fetchData = async (updateCompleteGenresList: (genres: Genre[]) => void) => {
+  try {
+    const completeGenresList = await fetchGenres({})
```
Comment (d-reader-josip) [inline src/providers/DiscoverFilterStoreProvider.tsx:15]: remove empty {} from `fetchGenres({})`

File: src/app/lib/api/genre/queries.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
+'use server'
+
+import { fetchWrapper } from '../../fetchWrapper'
+import { Genre } from '@/models/genre'
+import { GenreParams } from '@/models/genre/genreParams'
+import { GENRE_QUERY_KEYS } from '@/api/genre/genreKeys'
+
+const { GENRE, GET } = GENRE_QUERY_KEYS
+
+export const fetchGenres = async (params: GenreParams): Promise<Genre[]> => {
```
Comment (d-reader-josip) [inline src/app/lib/api/genre/queries.ts:10]: make params optional `(params?: GenreParams)`

**Review by d-reader-josip — CHANGES_REQUESTED** [review]

**Pull request merged**